Compile error on Ubuntu 20.04

I just run following commands
```
git clone https://github.com/RainerKuemmerle/g2o.git
mkdir build
cd build
cmake ../
make
```

and errors occur
```
[  4%] Built target stuff
[ 16%] Built target core
[ 18%] Built target g2o_cli_library
[ 19%] Built target g2o_cli_application
[ 28%] Built target types_slam3d
[ 32%] Built target g2o_hierarchical_library
[ 33%] Built target g2o_hierarchical_application
[ 40%] Built target types_slam2d
[ 44%] Built target types_data
[ 53%] Built target types_sba
[ 54%] Built target types_sim3
[ 55%] Built target types_icp
[ 58%] Built target types_sclam2d
[ 63%] Built target types_slam2d_addons
[ 68%] Built target types_slam3d_addons
[ 69%] Built target solver_pcg
[ 70%] Built target solver_dense
[ 71%] Built target solver_eigen
[ 72%] Built target solver_slam2d_linear
[ 73%] Built target solver_structure_only
[ 74%] Built target ba_demo
[ 74%] Built target ba_anchored_inverse_depth_demo
[ 75%] Building CXX object g2o/examples/bal/CMakeFiles/bal_example.dir/bal_example.cpp.o
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:96:52: error: template argument 2 is invalid
   96 | class VertexPointBAL : public BaseVertex<3, Vector3> {
      |                                                    ^
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp: In member function ‘virtual void VertexPointBAL::oplusImpl(const double*)’:
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:114:5: error: reference to ‘Vector3’ is ambiguous
  114 |     Vector3::ConstMapType v(update);
      |     ^~~~~~~
In file included from /usr/local/include/eigen3/Eigen/Core:295,
                 from /home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:27:
/usr/local/include/eigen3/Eigen/src/Core/Matrix.h:524:7: note: candidates are: ‘template<class Type> using Vector3 = Eigen::Matrix<Type, 3, 1>’
  524 | using Vector##SizeSuffix = Matrix<Type, Size, 1>;                 \
      |       ^~~~~~
/usr/local/include/eigen3/Eigen/src/Core/Matrix.h:541:1: note: in expansion of macro ‘EIGEN_MAKE_TYPEDEFS’
  541 | EIGEN_MAKE_TYPEDEFS(3, 3)
      | ^~~~~~~~~~~~~~~~~~~
In file included from /home/xuqi/pkgs/g2o/g2o/core/auto_differentiation.h:34,
                 from /home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:32:
/home/xuqi/pkgs/g2o/g2o/core/eigen_types.h:50:9: note:                 ‘using Vector3 = g2o::VectorN<3>’
   50 |   using Vector3 = VectorN<3>;
      |         ^~~~~~~
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:115:5: error: ‘_estimate’ was not declared in this scope
  115 |     _estimate += v;
      |     ^~~~~~~~~
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:115:18: error: ‘v’ was not declared in this scope
  115 |     _estimate += v;
      |                  ^
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:113:40: warning: unused parameter ‘update’ [-Wunused-parameter]
  113 |   virtual void oplusImpl(const double* update) {
      |                          ~~~~~~~~~~~~~~^~~~~~
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp: At global scope:
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:141:93: error: template argument 2 is invalid
  141 | class EdgeObservationBAL : public BaseBinaryEdge<2, Vector2, VertexCameraBAL, VertexPointBAL> {
      |                                                                                             ^
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp: In member function ‘bool EdgeObservationBAL::operator()(const T*, const T*, T*) const’:
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:197:26: error: there are no arguments to ‘measurement’ that depend on a template parameter, so a declaration of ‘measurement’ must be available [-fpermissive]
  197 |     error = prediction - measurement().cast<T>();
      |                          ^~~~~~~~~~~
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:197:26: note: (if you use ‘-fpermissive’, G++ will accept your code, but allowing the use of an undeclared name is deprecated)
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:197:46: error: expected primary-expression before ‘>’ token
  197 |     error = prediction - measurement().cast<T>();
      |                                              ^
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:197:48: error: expected primary-expression before ‘)’ token
  197 |     error = prediction - measurement().cast<T>();
      |                                                ^
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp: In function ‘int main(int, char**)’:
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:276:10: error: ‘class VertexPointBAL’ has no member named ‘setId’
  276 |       p->setId(id);
      |          ^~~~~
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:277:10: error: ‘class VertexPointBAL’ has no member named ‘setMarginalized’
  277 |       p->setMarginalized(true);
      |          ^~~~~~~~~~~~~~~
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:278:47: error: no matching function for call to ‘g2o::SparseOptimizer::addVertex(VertexPointBAL*&)’
  278 |       bool addedVertex = optimizer.addVertex(p);
      |                                               ^
In file included from /home/xuqi/pkgs/g2o/g2o/core/base_edge.h:35,
                 from /home/xuqi/pkgs/g2o/g2o/core/base_fixed_sized_edge.h:35,
                 from /home/xuqi/pkgs/g2o/g2o/core/base_binary_edge.h:30,
                 from /home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:33:
/home/xuqi/pkgs/g2o/g2o/core/optimizable_graph.h:572:18: note: candidate: ‘virtual bool g2o::OptimizableGraph::addVertex(g2o::HyperGraph::Vertex*, g2o::HyperGraph::Data*)’
  572 |     virtual bool addVertex(HyperGraph::Vertex* v, Data* userData);
      |                  ^~~~~~~~~
/home/xuqi/pkgs/g2o/g2o/core/optimizable_graph.h:572:18: note:   candidate expects 2 arguments, 1 provided
/home/xuqi/pkgs/g2o/g2o/core/optimizable_graph.h:573:18: note: candidate: ‘virtual bool g2o::OptimizableGraph::addVertex(g2o::HyperGraph::Vertex*)’
  573 |     virtual bool addVertex(HyperGraph::Vertex* v) { return addVertex(v, 0);}
      |                  ^~~~~~~~~
/home/xuqi/pkgs/g2o/g2o/core/optimizable_graph.h:573:48: note:   no known conversion for argument 1 from ‘VertexPointBAL*’ to ‘g2o::HyperGraph::Vertex*’
  573 |     virtual bool addVertex(HyperGraph::Vertex* v) { return addVertex(v, 0);}
      |                            ~~~~~~~~~~~~~~~~~~~~^
/home/xuqi/pkgs/g2o/g2o/core/optimizable_graph.h:574:10: note: candidate: ‘bool g2o::OptimizableGraph::addVertex(g2o::OptimizableGraph::Vertex*, g2o::HyperGraph::Data*)’
  574 |     bool addVertex(OptimizableGraph::Vertex* v, Data* userData);
      |          ^~~~~~~~~
/home/xuqi/pkgs/g2o/g2o/core/optimizable_graph.h:574:10: note:   candidate expects 2 arguments, 1 provided
/home/xuqi/pkgs/g2o/g2o/core/optimizable_graph.h:575:10: note: candidate: ‘bool g2o::OptimizableGraph::addVertex(g2o::OptimizableGraph::Vertex*)’
  575 |     bool addVertex(OptimizableGraph::Vertex* v) { return addVertex(v, 0); }
      |          ^~~~~~~~~
/home/xuqi/pkgs/g2o/g2o/core/optimizable_graph.h:575:46: note:   no known conversion for argument 1 from ‘VertexPointBAL*’ to ‘g2o::OptimizableGraph::Vertex*’
  575 |     bool addVertex(OptimizableGraph::Vertex* v) { return addVertex(v, 0); }
      |                    ~~~~~~~~~~~~~~~~~~~~~~~~~~^
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:297:10: error: ‘class EdgeObservationBAL’ has no member named ‘setVertex’
  297 |       e->setVertex(0, cam);
      |          ^~~~~~~~~
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:298:10: error: ‘class EdgeObservationBAL’ has no member named ‘setVertex’
  298 |       e->setVertex(1, point);
      |          ^~~~~~~~~
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:299:10: error: ‘class EdgeObservationBAL’ has no member named ‘setInformation’
  299 |       e->setInformation(Matrix2::Identity());
      |          ^~~~~~~~~~~~~~
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:299:25: error: reference to ‘Matrix2’ is ambiguous
  299 |       e->setInformation(Matrix2::Identity());
      |                         ^~~~~~~
In file included from /usr/local/include/eigen3/Eigen/Core:295,
                 from /home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:27:
/usr/local/include/eigen3/Eigen/src/Core/Matrix.h:520:7: note: candidates are: ‘template<class Type> using Matrix2 = Eigen::Matrix<Type, 2, 2>’
  520 | using Matrix##SizeSuffix = Matrix<Type, Size, Size>;              \
      |       ^~~~~~
/usr/local/include/eigen3/Eigen/src/Core/Matrix.h:540:1: note: in expansion of macro ‘EIGEN_MAKE_TYPEDEFS’
  540 | EIGEN_MAKE_TYPEDEFS(2, 2)
      | ^~~~~~~~~~~~~~~~~~~
In file included from /home/xuqi/pkgs/g2o/g2o/core/auto_differentiation.h:34,
                 from /home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:32:
/home/xuqi/pkgs/g2o/g2o/core/eigen_types.h:68:9: note:                 ‘using Matrix2 = g2o::MatrixN<2>’
   68 |   using Matrix2 = MatrixN<2>;
      |         ^~~~~~~
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:300:10: error: ‘class EdgeObservationBAL’ has no member named ‘setMeasurement’
  300 |       e->setMeasurement(Vector2(obsX, obsY));
      |          ^~~~~~~~~~~~~~
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:300:25: error: reference to ‘Vector2’ is ambiguous
  300 |       e->setMeasurement(Vector2(obsX, obsY));
      |                         ^~~~~~~
In file included from /usr/local/include/eigen3/Eigen/Core:295,
                 from /home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:27:
/usr/local/include/eigen3/Eigen/src/Core/Matrix.h:524:7: note: candidates are: ‘template<class Type> using Vector2 = Eigen::Matrix<Type, 2, 1>’
  524 | using Vector##SizeSuffix = Matrix<Type, Size, 1>;                 \
      |       ^~~~~~
/usr/local/include/eigen3/Eigen/src/Core/Matrix.h:540:1: note: in expansion of macro ‘EIGEN_MAKE_TYPEDEFS’
  540 | EIGEN_MAKE_TYPEDEFS(2, 2)
      | ^~~~~~~~~~~~~~~~~~~
In file included from /home/xuqi/pkgs/g2o/g2o/core/auto_differentiation.h:34,
                 from /home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:32:
/home/xuqi/pkgs/g2o/g2o/core/eigen_types.h:49:9: note:                 ‘using Vector2 = g2o::VectorN<2>’
   49 |   using Vector2 = VectorN<2>;
      |         ^~~~~~~
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:301:43: error: no matching function for call to ‘g2o::SparseOptimizer::addEdge(EdgeObservationBAL*&)’
  301 |       bool addedEdge = optimizer.addEdge(e);
      |                                           ^
In file included from /home/xuqi/pkgs/g2o/g2o/core/base_edge.h:35,
                 from /home/xuqi/pkgs/g2o/g2o/core/base_fixed_sized_edge.h:35,
                 from /home/xuqi/pkgs/g2o/g2o/core/base_binary_edge.h:30,
                 from /home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:33:
/home/xuqi/pkgs/g2o/g2o/core/optimizable_graph.h:582:18: note: candidate: ‘virtual bool g2o::OptimizableGraph::addEdge(g2o::HyperGraph::Edge*)’
  582 |     virtual bool addEdge(HyperGraph::Edge* e);
      |                  ^~~~~~~
/home/xuqi/pkgs/g2o/g2o/core/optimizable_graph.h:582:44: note:   no known conversion for argument 1 from ‘EdgeObservationBAL*’ to ‘g2o::HyperGraph::Edge*’
  582 |     virtual bool addEdge(HyperGraph::Edge* e);
      |                          ~~~~~~~~~~~~~~~~~~^
/home/xuqi/pkgs/g2o/g2o/core/optimizable_graph.h:583:10: note: candidate: ‘bool g2o::OptimizableGraph::addEdge(g2o::OptimizableGraph::Edge*)’
  583 |     bool addEdge(OptimizableGraph::Edge* e);
      |          ^~~~~~~
/home/xuqi/pkgs/g2o/g2o/core/optimizable_graph.h:583:42: note:   no known conversion for argument 1 from ‘EdgeObservationBAL*’ to ‘g2o::OptimizableGraph::Edge*’
  583 |     bool addEdge(OptimizableGraph::Edge* e);
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~^
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:317:7: error: reference to ‘Vector3’ is ambiguous
  317 |       Vector3 p;
      |       ^~~~~~~
In file included from /usr/local/include/eigen3/Eigen/Core:295,
                 from /home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:27:
/usr/local/include/eigen3/Eigen/src/Core/Matrix.h:524:7: note: candidates are: ‘template<class Type> using Vector3 = Eigen::Matrix<Type, 3, 1>’
  524 | using Vector##SizeSuffix = Matrix<Type, Size, 1>;                 \
      |       ^~~~~~
/usr/local/include/eigen3/Eigen/src/Core/Matrix.h:541:1: note: in expansion of macro ‘EIGEN_MAKE_TYPEDEFS’
  541 | EIGEN_MAKE_TYPEDEFS(3, 3)
      | ^~~~~~~~~~~~~~~~~~~
In file included from /home/xuqi/pkgs/g2o/g2o/core/auto_differentiation.h:34,
                 from /home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:32:
/home/xuqi/pkgs/g2o/g2o/core/eigen_types.h:50:9: note:                 ‘using Vector3 = g2o::VectorN<3>’
   50 |   using Vector3 = VectorN<3>;
      |         ^~~~~~~
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:318:14: error: ‘p’ was not declared in this scope
  318 |       ifs >> p(0) >> p(1) >> p(2);
      |              ^
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:320:14: error: ‘class VertexPointBAL’ has no member named ‘setEstimate’
  320 |       point->setEstimate(p);
      |              ^~~~~~~~~~~
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:359:22: error: ‘class VertexPointBAL’ has no member named ‘estimate’
  359 |       fout << (*it)->estimate().transpose() << endl;
      |                      ^~~~~~~~
In file included from /home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:32:
/home/xuqi/pkgs/g2o/g2o/core/auto_differentiation.h: In instantiation of ‘static void g2o::AutoDifferentiation<Edge, EstimateAccess>::computeError(Edge*) [with Edge = EdgeObservationBAL; EstimateAccess = g2o::EstimateAccessor<EdgeObservationBAL>]’:
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:202:3:   required from here
/home/xuqi/pkgs/g2o/g2o/core/auto_differentiation.h:156:35: error: ‘Dimension’ is not a member of ‘EdgeObservationBAL’
  156 |     static_assert(Edge::Dimension > 0, "Dynamically sized edges are not supported");
      |                         ~~~~~~~~~~^~~
/home/xuqi/pkgs/g2o/g2o/core/auto_differentiation.h:157:31: error: ‘_nr_of_vertices’ is not a member of ‘EdgeObservationBAL’
  157 |     computeErrorNs(that, std::make_index_sequence<Edge::_nr_of_vertices>());
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/xuqi/pkgs/g2o/g2o/core/auto_differentiation.h: In instantiation of ‘static void g2o::AutoDifferentiation<Edge, EstimateAccess>::linearize(Edge*) [with Edge = EdgeObservationBAL; EstimateAccess = g2o::EstimateAccessor<EdgeObservationBAL>]’:
/home/xuqi/pkgs/g2o/g2o/examples/bal/bal_example.cpp:202:3:   required from here
/home/xuqi/pkgs/g2o/g2o/core/auto_differentiation.h:168:35: error: ‘Dimension’ is not a member of ‘EdgeObservationBAL’
  168 |     static_assert(Edge::Dimension > 0, "Dynamically sized edges are not supported");
      |                         ~~~~~~~~~~^~~
/home/xuqi/pkgs/g2o/g2o/core/auto_differentiation.h:169:33: error: ‘_nr_of_vertices’ is not a member of ‘EdgeObservationBAL’
  169 |     linearizeOplusNs(that, std::make_index_sequence<Edge::_nr_of_vertices>());
      |                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [g2o/examples/bal/CMakeFiles/bal_example.dir/build.make:63：g2o/examples/bal/CMakeFiles/bal_example.dir/bal_example.cpp.o] error 1
make[1]: *** [CMakeFiles/Makefile2:1652：g2o/examples/bal/CMakeFiles/bal_example.dir/all] error 2
make: *** [Makefile:152：all] error 2
```

I am so confused !
does anyone can help me?

--- a/g2o/examples/bal/bal_example.cpp
+++ b/g2o/examples/bal/bal_example.cpp
@@ -48,7 +48,6 @@
 
 using namespace g2o;
 using namespace std;
-using namespace Eigen;
 
 namespace g2o {
 namespace bal {
@@ -195,7 +194,7 @@ class EdgeObservationBAL : public BaseBinaryEdge<2, Vector2, VertexCameraBAL, Ve
     // compute the error
     typename g2o::VectorN<2, T>::MapType error(p_error);
     error = prediction - measurement().cast<T>();
-    (void) error;
+    (void)error;
     return true;
   }
 

--- a/g2o/examples/bal/bal_example.cpp
+++ b/g2o/examples/bal/bal_example.cpp
@@ -46,7 +46,6 @@
 #include "g2o/solvers/eigen/linear_solver_eigen.h"
 #endif
 
-using namespace g2o;
 using namespace std;
 
 namespace g2o {
@@ -64,7 +63,7 @@ using Vector9 = VectorN<9>;
  * - f the focal length of the camera
  * - k1, k2 two radial distortion parameters
  */
-class VertexCameraBAL : public BaseVertex<9, bal::Vector9> {
+class VertexCameraBAL : public g2o::BaseVertex<9, g2o::bal::Vector9> {
  public:
   EIGEN_MAKE_ALIGNED_OPERATOR_NEW;
   VertexCameraBAL() {}
@@ -82,7 +81,7 @@ class VertexCameraBAL : public BaseVertex<9, bal::Vector9> {
   virtual void setToOriginImpl() { cerr << __PRETTY_FUNCTION__ << " not implemented yet" << endl; }
 
   virtual void oplusImpl(const double* update) {
-    bal::Vector9::ConstMapType v(update, VertexCameraBAL::Dimension);
+    g2o::bal::Vector9::ConstMapType v(update, VertexCameraBAL::Dimension);
     _estimate += v;
   }
 };
@@ -92,7 +91,7 @@ class VertexCameraBAL : public BaseVertex<9, bal::Vector9> {
  *
  * A 3D point feature in the world
  */
-class VertexPointBAL : public BaseVertex<3, Vector3> {
+class VertexPointBAL : public g2o::BaseVertex<3, g2o::Vector3> {
  public:
   EIGEN_MAKE_ALIGNED_OPERATOR_NEW;
   VertexPointBAL() {}
@@ -110,7 +109,7 @@ class VertexPointBAL : public BaseVertex<3, Vector3> {
   virtual void setToOriginImpl() { cerr << __PRETTY_FUNCTION__ << " not implemented yet" << endl; }
 
   virtual void oplusImpl(const double* update) {
-    Vector3::ConstMapType v(update);
+    g2o::Vector3::ConstMapType v(update);
     _estimate += v;
   }
 };
@@ -137,7 +136,8 @@ class VertexPointBAL : public BaseVertex<3, Vector3> {
  * z-axis points backwards, so the camera is looking down the negative z-axis,
  * as in OpenGL).
  */
-class EdgeObservationBAL : public BaseBinaryEdge<2, Vector2, VertexCameraBAL, VertexPointBAL> {
+class EdgeObservationBAL
+    : public g2o::BaseBinaryEdge<2, g2o::Vector2, VertexCameraBAL, VertexPointBAL> {
  public:
   EIGEN_MAKE_ALIGNED_OPERATOR_NEW;
   EdgeObservationBAL() {}
@@ -208,7 +208,7 @@ int main(int argc, char** argv) {
   string outputFilename;
   string inputFilename;
   string statsFilename;
-  CommandArgs arg;
+  g2o::CommandArgs arg;
   arg.param("i", maxIterations, 5, "perform n iterations");
   arg.param("o", outputFilename, "", "write points into a vrml file");
   arg.param("pcg", usePCG, false, "use PCG instead of the Cholesky");
@@ -295,8 +295,8 @@ int main(int argc, char** argv) {
       EdgeObservationBAL* e = new EdgeObservationBAL;
       e->setVertex(0, cam);
       e->setVertex(1, point);
-      e->setInformation(Matrix2::Identity());
-      e->setMeasurement(Vector2(obsX, obsY));
+      e->setInformation(g2o::Matrix2::Identity());
+      e->setMeasurement(g2o::Vector2(obsX, obsY));
       bool addedEdge = optimizer.addEdge(e);
       if (!addedEdge) {
         cerr << "error adding edge" << endl;
@@ -305,7 +305,7 @@ int main(int argc, char** argv) {
 
     // read in the camera params
     for (int i = 0; i < numCameras; ++i) {
-      bal::Vector9 cameraParameter;
+      g2o::bal::Vector9 cameraParameter;
       for (int j = 0; j < 9; ++j) ifs >> cameraParameter(j);
       VertexCameraBAL* cam = cameras[i];
       cam->setEstimate(cameraParameter);
@@ -313,7 +313,7 @@ int main(int argc, char** argv) {
 
     // read in the points
     for (int i = 0; i < numPoints; ++i) {
-      Vector3 p;
+      g2o::Vector3 p;
       ifs >> p(0) >> p(1) >> p(2);
       VertexPointBAL* point = points[i];
       point->setEstimate(p);
@@ -331,7 +331,7 @@ int main(int argc, char** argv) {
   if (statsFilename != "") {
     cerr << "writing stats to file \"" << statsFilename << "\" ... ";
     ofstream fout(statsFilename.c_str());
-    const BatchStatisticsContainer& bsc = optimizer.batchStatistics();
+    const g2o::BatchStatisticsContainer& bsc = optimizer.batchStatistics();
     for (size_t i = 0; i < bsc.size(); i++) fout << bsc[i] << endl;
     cerr << "done." << endl;
   }

